pkg/validators/validate_test_harness.go

```diff
@@ -0,0 +1,43 @@
+// ValidateAddonLabel validates whether the 'label' field under an addon.yaml follows the format 'api.openshift.com/addon-<id>'
+// TODO - remove this validator once we do field level validation
+package validators
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/mt-sre/addon-metadata-operator/pkg/utils"
+	imageparser "github.com/novln/docker-parser"
+)
+
+const quayRegistryApi = "https://quay.io/v2"
+
+func ValidateTestHarness(metabundle utils.MetaBundle) (bool, string, error) {
+	res, err := imageparser.Parse(metabundle.AddonMeta.TestHarness)
+	if err != nil {
+		return false, "Failed to parse testharness url", err
+	}
+	if res.Registry() != "quay.io" {
+		return false, "Testharness image is not in the quay.io registry", nil
+	}
+	imagePresent, err := checkImageExists(res)
+	if err != nil {
+		return false, "Encountered an error when trying to access the remote image", err
+	}
+	if !imagePresent {
+		return false, "Test harness image doesnt exist", nil
+	}
+	return true, "", nil
+}
+
+func checkImageExists(imageUri *imageparser.Reference) (bool, error) {
```

Comment:
As `quay.io` has proven quite flaky recently, do you think it's worth adding a retry mechanism for more reliability?